src/main/java/org/elasticsearch/common/settings/SettingsFilter.java

```diff
@@ -18,41 +18,95 @@
  */
 package org.elasticsearch.common.settings;
 
+import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.rest.RestRequest;
 
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
 import java.util.concurrent.CopyOnWriteArrayList;
 
+import static com.google.common.collect.Lists.newArrayList;
+
 /**
  *
  */
 public class SettingsFilter extends AbstractComponent {
```

Comment:
this class seems to be easily unittestable? Can we have a pure unittest for this that tests all the concurrency aspects of the add / remove as well as the modifiers?